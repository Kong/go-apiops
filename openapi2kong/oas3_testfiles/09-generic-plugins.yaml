# x-kong-plugin-<name> are accepted on each level. The document level plugins
# end up on the service. The path and operation level plugins on the route.
#
# if the same plugin is on a path and operation, the operation one wins (to be
# attached to a route).
#
# Plugin-defaults cannot have foreign-keys to service/route entities (ignored)

openapi: '3.0.0'
info:
  title: Simple API overview
  version: v2
servers:
  - url: https://server1.com/

x-kong-plugin-request-termination:
  # name will be determined from the tag
  # name: request-termination;
  config:
    status_code: 403
    message: So long and thanks for all the fish!
  # foreign keys to be ignored
  service: delete-me
  route: delete-me-to

paths:
  /path1:
    get:
      # This operation should not get a plugin, but rely on the document level
      # service-entity based plugin
      operationId: uses-doc-plugin
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
  /path2:
    # This plugin gets attached to a route, overriding the document service-entity
    # attached plugin
    x-kong-plugin-request-termination:
      config:
        status_code: 403
        message: The answer to life, the universe, and everything!
      # foreign keys to be ignored
      service: delete-me
      route: delete-me-to
    get:
      # gets the plugin from the path-level, attached on route-entity
      operationId: uses-path-plugin
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
    post:
      # This plugin gets attached to a route, taking precedence over the path-level
      # plugin. On the route it will override the document level service-entity
      # attached plugin
      x-kong-plugin-request-termination:
        config:
          status_code: 403
          message: For a moment, nothing happened. Then, after a second or so, nothing continued to happen.
        # foreign keys to be ignored
        service: delete-me
        route: delete-me-to
      operationId: uses-ops-plugin
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
