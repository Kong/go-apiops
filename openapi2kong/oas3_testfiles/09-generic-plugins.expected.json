{
  "_format_version": "3.0",
  "services": [
    {
      "host": "server1.com",
      "id": "e3a95664-95d7-5b10-bc3d-11c9cc6c2b4e",
      "name": "Simple_API_overview",
      "path": "/",
      "plugins": [
        {
          "config": {
            "message": "So long and thanks for all the fish!",
            "status_code": 403
          },
          "id": "3a0a598e-5e31-570d-abcc-563876fa0a0b",
          "name": "request-termination",
          "tags": [
            "OAS3_import",
            "OAS3file_09-generic-plugins.yaml"
          ]
        }
      ],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "8062e64f-dce4-5ae1-aa55-b2c3a1ae9422",
          "methods": [
            "GET"
          ],
          "name": "Simple_API_overview-uses-doc-plugin",
          "paths": [
            "~/path1$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_09-generic-plugins.yaml"
          ]
        },
        {
          "id": "f875b5f9-2e16-52ef-9f83-a407a1b73a32",
          "methods": [
            "GET"
          ],
          "name": "Simple_API_overview-uses-path-plugin",
          "paths": [
            "~/path2$"
          ],
          "plugins": [
            {
              "config": {
                "message": "The answer to life, the universe, and everything!",
                "status_code": 403
              },
              "id": "c518bbe3-0e8b-53bc-b845-733fc4e98fe8",
              "name": "request-termination",
              "tags": [
                "OAS3_import",
                "OAS3file_09-generic-plugins.yaml"
              ]
            }
          ],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_09-generic-plugins.yaml"
          ]
        },
        {
          "id": "32e8ee5a-df88-5e35-888d-ef404543a91d",
          "methods": [
            "POST"
          ],
          "name": "Simple_API_overview-uses-ops-plugin",
          "paths": [
            "~/path2$"
          ],
          "plugins": [
            {
              "config": {
                "message": "For a moment, nothing happened. Then, after a second or so, nothing continued to happen.",
                "status_code": 403
              },
              "id": "c2294326-c1e8-55ae-8f2f-e1a612556beb",
              "name": "request-termination",
              "tags": [
                "OAS3_import",
                "OAS3file_09-generic-plugins.yaml"
              ]
            }
          ],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_09-generic-plugins.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_09-generic-plugins.yaml"
      ]
    }
  ],
  "upstreams": []
}